pandas/tests/io/formats/test_format.py

```diff
@@ -3167,6 +3166,65 @@ def test_str(self):
         assert str(NaT) == "NaT"
 
 
+class TestPeriodIndexFormat:
+    def test_period(self):
+        """Basic test for period formatting with default format."""
+        p = pd.PeriodIndex([datetime(2003, 1, 1, 12), None], freq="H")
```

Comment:
more nitpicks: 1) pls avoid 1-letter variable names, 2) generally dont use docstrings for tests

Reply:
comments for the most part.  im particular to the form `# GH#12345 case where [...]`

Reply:
ok